Bonsai.Tinkerforge/BrickletDeviceNameConverter.cs

```diff
@@ -0,0 +1,158 @@
+        /// and what UIDs are present on those connections
+        /// </summary>
+        public override StandardValuesCollection GetStandardValues(ITypeDescriptorContext context)
+        {
+            if (context != null)
+            {
+                // We need to know which CreateIPConnection nodes are present, to know what channels we should look for devices on
+                // Therefore we search the workflow for CreateIPConnection nodes. From each node we extract the Host / Port
+                var workflowBuilder = (WorkflowBuilder)context.GetService(typeof(WorkflowBuilder));
+                if (workflowBuilder != null)
+                {
+                    var connectionIDs = (from builder in workflowBuilder.Workflow.Descendants()
+                                         let createIP = ExpressionBuilder.GetWorkflowElement(builder) as CreateIPConnection
+                                         where createIP != null
+                                         select new ConnectionID { HostName = createIP.HostName, Port = createIP.Port })
+                                         .Distinct().ToList();
+
+                    // For each IP connection, we search the connected devices and add it to the list - TODO test for multiple IP
+                    devices = new Dictionary<string, TinkerforgeHelpers.DeviceData>();
+                    foreach (ConnectionID connectionID in connectionIDs)
+                    {
+                        var ipcon = new IPConnection();
+                        ipcon.EnumerateCallback += EnumerateConnection;
+                        ipcon.Connect(connectionID.HostName, connectionID.Port); 
+                        ipcon.Enumerate();
+                        /// N.B. GetStandardValues is called twice by Windows Forms. Once to check the dropdown list and then again
+                        /// to check cursor position in the list. This is annoying here because it means the enumerate thread will
+                        /// be called twice leading to e.g. the device list being cleared in the middle of population. The 'hack' 
+                        /// here is to sleep the thread so it can complete before the 2nd call. Not optimal but the Enumerate 
+                        /// method doesn't give us much choice. TODO - look for more elegant solution
+                        Thread.Sleep(10);
+                        ipcon.Disconnect();
+                    }
+                }
+                //return new StandardValuesCollection(devices.Select(x => x.DeviceIdentifier).ToList());
+                return new StandardValuesCollection(devices.Values.ToList());
+            }
+
+            return base.GetStandardValues(context);
+        }
+
+        /// <summary>
+        /// EnumerateCallback used to add devices to the device list when IPConnection.Enumerate() is called
+        /// </summary>
```

Comment:
Can we replace this by inline comments inside the method implementation itself? We usually reserve the use of `<summary>` tags for public reference documentation.